core/src/test/java/kafka/server/BootstrapControllersIntegrationTest.java

```diff
@@ -228,6 +228,17 @@ public void testIncrementalAlterConfigs(ClusterInstance clusterInstance) throws
         testIncrementalAlterConfigs(clusterInstance, false);
     }
 
+    @ClusterTest(brokers = 2)
+    public void testUnregisterBroker(ClusterInstance clusterInstance) throws Exception {
```

Comment:
Could you please consider integrating `KRaftClusterTest#testUnregisterBroker` into this test?